server/src/com/cloud/network/router/VpcVirtualNetworkApplianceManagerImpl.java

```diff
@@ -260,6 +261,15 @@ public boolean finalizeVirtualMachineProfile(final VirtualMachineProfile profile
                 if (defaultDns2 != null) {
                     buf.append(" dns2=").append(defaultDns2);
                 }
+
+                VpcGatewayVO privateGatewayForVpc = _vpcGatewayDao.getPrivateGatewayForVpc(domainRouterVO.getVpcId());
+                if (privateGatewayForVpc != null) {
+                    String ip4Address = privateGatewayForVpc.getIp4Address();
+                    buf.append(" privategateway=").append(ip4Address);
+                    s_logger.debug("Set privategateway field in cmd_line.json to " + ip4Address);
+                } else {
+                    buf.append(" privategateway=None");
+                }
```

Comment:
matter of style: I would have assigned "None" to ip4Assress and appended the buffer outside the conditional and maybe factorred the block out as method. no issue.